### PR TITLE
[dashboard] always render context URL without prefixes

### PR DESCRIPTION
**Pull request by akosyakov**
## Description

Right now:
- some URLs are rendered with prefixes on UI
- there is code duplicating prefix parsing of context URIs used on UI

This PR assumes removes duplicated code and always render context URLs without prefixes.

It is a breaking change, but I am not sure to which extend, i.e. are there really workspace without normlalized context URLs in DB? @geropl ?

Alternatively I can open another PR which uses normalized context URL or fallbacks to parsing but only for old prefix handlers for new we won't do it anymore.

## Related Issue(s)

For JetBrains we will use a special prefix handler so all workspaces started from JB Gateway will be started with it. This prefix leaks to UI here and there.

## How to test

- Start a workspace with prefix.
- Make sure that you don't see it anywhere on UI.

## Release Notes

```release-note
NONE
```

### CODE REVIEW

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **geropl** after the PR has been reviewed.
You can assign the PR to them by writing `/assign @geropl` in a comment when ready.

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

<details open>
Needs approval from an approver in each of these files:

- **[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)**
- **[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)**
- **[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: components/gitpod-protocol/src/protocol.ts
Comment (akosyakov) [inline components/gitpod-protocol/src/protocol.ts:779]: revert this line

**Review by gtsiolis — APPROVED** [review]
UX changes work as advertized! Thanks for the ping @akosyakov! 🔮

Approving to unblock merging but holding to give time to other reviewers to take a closer look at the code changes.

/hold

**Comment by roboquat** [comment]
LGTM label has been added.  <details>Git tree hash: 09040e9b6bdf22d5eaa147e2f3d9c73bc32c9552</details>

**Comment by roboquat** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by: *<a href="https://github.com/gitpod-io/gitpod/pull/7617#pullrequestreview-854347374" title="Approved">gtsiolis</a>*

*No associated issue*. Update pull-request body to add a reference to an issue, or get approval with `/approve no-issue`

The full list of commands accepted by this bot can be found [here](https://prow.gitpod-dev.com/command-help?repo=gitpod-io%2Fgitpod).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[components/dashboard/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/dashboard/OWNERS)~~ [gtsiolis]
- ~~[components/gitpod-protocol/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/gitpod-protocol/OWNERS)~~ [gtsiolis]
- ~~[components/server/OWNERS](https://github.com/gitpod-io/gitpod/blob/main/components/server/OWNERS)~~ [gtsiolis]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by akosyakov** [comment]
@gtsiolis thanks, someone should look whether it is backward compatible or rendering of old workspaces can be broken with it

File: components/gitpod-protocol/src/protocol.ts
Comment (geropl) [inline components/gitpod-protocol/src/protocol.ts:403]: :+1:

**Comment by geropl** [comment]
@akosyakov I checked the DB; quite a lot workspaces do not have `normaliedContextURL` set. Maybe we should fix that first; otherwise even with `p.ws.context.normalizedContextURL || p.ws.contextURL` this looks weird. Once that's fixed we can merge this change with the fallback to `contextURL`. WDYT?

**Comment by geropl** [comment]
Issue: https://github.com/gitpod-io/gitpod/issues/7681

**Comment by akosyakov** [comment]
> I checked the DB; quite a lot workspaces do not have normaliedContextURL set. Maybe we should fix that first; otherwise even with p.ws.context.normalizedContextURL || p.ws.contextURL this looks weird. Once that's fixed we can merge this change with the fallback to contextURL. WDYT?

Instead of removing `parseURL` I can change it in a way that it first checks `normalizedContextURL` and add a comment that we should not reimplement the logic here for new prefix handlers. After that update UI code to use `parseURL` consistently and add a comment to `Workspace.contextURL` to use it instead for rendering. In such way you don't need to fix records in DB?

**Comment by akosyakov** [comment]
Closing it for now, since I don't really have time to look into it.

**Pull request closed without merging**
